src/chart/graph/categoryVisual.js

```diff
@@ -31,30 +31,53 @@ export default function (ecModel) {
             var name = categoriesData.getName(idx);
             // Add prefix to avoid conflict with Object.prototype.
             categoryNameIdxMap['ec-' + name] = idx;
-
             var itemModel = categoriesData.getItemModel(idx);
+
             var color = itemModel.get('itemStyle.color')
                 || seriesModel.getColorFromPalette(name, paletteScope);
             categoriesData.setItemVisual(idx, 'color', color);
+
+            var opacity = itemModel.get('itemStyle.opacity') || 1;
+            categoriesData.setItemVisual(idx, 'opacity', opacity);
+
+            var symbolStyleList = ['symbolSize', 'symbolKeepAspect'];
+            for (var i = 0; i < symbolStyleList.length; i++) {
+                var symbolStyleItem = itemModel.getShallow(symbolStyleList[i], true);
+                if (symbolStyleItem != null) {
+                    categoriesData.setItemVisual(idx, symbolStyleList[i], symbolStyleItem);
+                }
+            }
+
         });
 
         // Assign category color to visual
         if (categoriesData.count()) {
             data.each(function (idx) {
+                var itemModel = categoriesData.getItemModel(idx);
+                var symbol = itemModel.getShallow('symbol', true);
+                if (symbol) {
```

Comment:
I prefer do not support customizing `symbol` in this case.
Because it's a little complicated. 
Set symbol to "data item visual" does not work.

Reply:
Actually it is what the issue for...
So I just leave it there...? o.o